packages/node-app/src/controllers/giveaway.controller.ts

```diff
@@ -41,7 +31,37 @@ export const listGiveaways = async (req: Request, res: Response) => {
       )
       .lean();
 
-    giveaways = giveaways.map((giveaway) => ({
+    const active = req.query.active === 'true' ? true : false;
```

Comment:
My understanding is that by default we always load on the `active` state, triggering its flag, the only other tab being the `archived` which is supposed to not contain active giveaways. So we are always requesting either active on non active giveaways. Is this correct?

Reply:
Yes, you're right, that's how the app works when showing the giveaways. I am just trying to avoid that coupling between frontend and backend. For someone working with this same api, if he tries to use this endpoint and don't fill the active parameter, he will receive the archived giveaways. I am just thinking what makes more sense, we have three possibilities:
- return the complete list
- return the active giveaways by default
- return the archived giveaways (what we currently have)